lib/node/runner/checker_ballot_transaction.go

```diff
@@ -282,14 +288,71 @@ func ValidateOp(st *storage.LevelDBBackend, source *block.BlockAccount, op trans
 			err = errors.ErrorFrozenAccountNoDeposit
 			return
 		}
-		// If it's a frozen account, everything must be withdrawn
 		if source.Linked != "" {
+			// If it's a frozen account, everything must be withdrawn
 			var expected common.Amount
 			expected, err = source.Balance.Sub(common.BaseFee)
 			if casted.Amount != expected {
 				err = errors.ErrorFrozenAccountMustWithdrawEverything
 				return
 			}
+			// Unfreezing must be done after X period from unfreezing request
+			iterFunc, closeFunc := block.GetBlockOperationsBySource(st, source.Address, nil)
+			bo, _, _ := iterFunc()
+			closeFunc()
+
+			var lastblock block.Block
+			lastblock, err = block.GetLatestBlock(st)
+			if err != nil {
+				return
+			}
+			if bo.Type != transaction.OperationUnfreezingRequest {
+				err = errors.ErrorUnfreezingRequestNotRequested
+				return
+			}
+
+			// unfreezing period is 241920.
+			if lastblock.Height-bo.BlockHeight <= common.UnfreezingPeriod {
+				err = errors.ErrorUnfreezingNotReachedExpiration
+				return
+			}
+		}
+	case transaction.OperationUnfreezingRequest:
+		var ok bool
+		var casted transaction.OperationBodyUnfreezeRequest
+		if casted, ok = op.B.(transaction.OperationBodyUnfreezeRequest); !ok {
+			err = errors.ErrorTypeOperationBodyNotMatched
+			return
+		}
+		// Unfreezing should be done from a frozen account
+		if source.Linked == "" {
+			err = errors.ErrorUnfreezingFromInvalidAccount
+			return
+		}
+		// Target account must be existed
+		var exists bool
+		if exists, err = block.ExistsBlockAccount(st, casted.TargetAddress()); err == nil && !exists {
+			err = errors.ErrorBlockAccountDoesNotExists
+			return
+		}
+		// Source account's linked address must be equal to target address
```

Comment:
Why ? I don't think that's necessary either

Reply:
@Geod24 I'll remove it. Thanks.